src/core/qgscopyfiletask.h

```diff
@@ -0,0 +1,58 @@
+/***************************************************************************
+  qgscopyfiletask.h
+  --------------------------------------
+  Date                 : March 2021
+  Copyright            : (C) 2021 by Julien Cabieces
+  Email                : julien dot cabieces at oslandia dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSCOPYFILETASK_H
+#define QGSCOPYFILETASK_H
+
+#include "qgstaskmanager.h"
+
+/**
+ * \ingroup core
+ * \brief Task to copy a file on disk
+ *
+ * \since QGIS 3.22
+ */
+class CORE_EXPORT QgsCopyFileTask : public QgsTask
```

Comment:
I'd like to see some specific tests for this class (outside of the storage tests)

Reply:
Done